tags/for.js

```diff
@@ -57,14 +61,47 @@ module.exports = function(liquid) {
                     stop: false,
                     skip: false
                 };
+                // todo: verify scope management is good here.  Make sure we don't consume too many resources here with the clone.
+                //  Is there a simpler solution?
                 scope.push(ctx);
-                html += liquid.renderer.renderTemplates(this.templates, scope);
-                var breakloop = scope.get('forloop.stop');
+                // We are just putting together an array of the arguments we will be passing to our sequential promises
+                scopes.push(_.clone(scope));
                 scope.pop(ctx);
-
-                if (breakloop) return true;
             });
-            return html;
+
+            // This is some pretty tricksy javascript, at least to me.  Bluebird would have made this a lot easier, but
+            //  we are trying to not use anything that can't be done in native node Promises.
+            // This basically just processes an array of promises sequentially for every argument in the array - http://webcache.googleusercontent.com/search?q=cache:rNbMUn9TPtkJ:joost.vunderink.net/blog/2014/12/15/processing-an-array-of-promises-sequentially-in-node-js/+&cd=5&hl=en&ct=clnk&gl=us
```

Comment:
you're right. We can extract these code and do polyfills to Promise later, to facilitate Promise upgrade.